troubleshooting content added (#2322)
SumoLogic/sumologic-documentation: docs/observability/aws/deploy-use-aws-observability/deploy-with-terraform.md

|An argument named **managed_apps** is not expected here.<br/>`Error: Unsupported argument`<br/>`on .terraform/modules/account.sumo_observability.app-modules/alb_app.tf line 13, in module "alb_module":`<br/>`managed_apps = {`<br/> | Refer the [module](https://github.com/SumoLogic/terraform-sumologic-sumo-logic-integrations/tree/master/sumologic).|
|An argument named **admin_mode** is not expected here.<br/>`Error: Unsupported argument`<br/>`on .terraform/modules/account.sumo_observability/provider.tf line 5, in provider "sumologic":`<br/>`admin_mode = var.sumologic_folder_installation_location == "Personal folder" ? false:true`<br/> | Sumologic provider [version 2.10.0](https://github.com/SumoLogic/terraform-provider-sumologic/blob/master/CHANGELOG.md#2100-september-22-2021) onwards supports `admin_mode` <br/>Refer the [admin_mode](https://registry.terraform.io/providers/SumoLogic/sumologic/latest/docs#authentication).|

> |Error: Invalid function argument<br/>`on.terraform/modules/sumo-module.overview_app.overview_module/sumologic/sumologic.tf line 67, in resource "sumologic_content" "SumoLogicApps":`<br/>`67: config = file(each.value.content_json)`<br/> | Verify [app JSONN location](https://github.com/SumoLogic/sumologic-solution-templates/tree/master/aws-observability/json) and align your custom tf script accordingly.|

Reviewer: ```suggestion
|Error: Invalid function argument<br/>`on.terraform/modules/sumo-module.overview_app.overview_module/sumologic/sumologic.tf line 67, in resource "sumologic_content" "SumoLogicApps":`<br/>`67: config = file(each.value.content_json)`<br/> | Verify [app JSON location](https://github.com/SumoLogic/sumologic-solution-templates/tree/master/aws-observability/json) and align your custom tf script accordingly.|
```

Author: fixed.

| `email_notifications`	Email Notifications to be sent by the alert. | `[ ]` |
| `connection_notifications` | Connection Notifications to be sent by the alert. | `[ ]` |
| `parent_folder_id` | The folder ID is automatically generated. Do not enter a value for this parameter. This is the folder ID to install the apps into. A folder using the provided name will be added in "apps_folder_name". If the folder ID is empty, apps will be installed in the Personal folder. | Ignore this parameter. |

## Troubleshooting

This section provides information on how to troubleshoot failures while deploying AWS observability solution using Terraform.

> ### 1. Python command not found

Reviewer: These headers don't need to be numbers, as it's just a list

> ### 1. Python command not found

Reviewer: These headers don't need to have numbers - it's just a list

```suggestion
### Python command not found
```

Author: Comments incorporated

#### Error Message
`python source-module/attach_fields_to_source.py`<br/>Python: command not found
#### Solution
Identify and replace `python` with `python3` in [source-module/update_sources.tf](https://github.com/SumoLogic/sumologic-solution-templates/blob/AWSO_FY23Q4_Release/aws-observability-terraform/source-module/update_sources.tf#L12).

### 2. Module not found
#### Error Message
`Local-exec provisioner error`<br/>Module Not Found Error: No Module named ‘sumologic’
#### Solution
Verify you configured [Sumo Logic provider](https://github.com/SumoLogic/sumologic-solution-templates/blob/AWSO_FY23Q4_Release/aws-observability-terraform/providers.tf#L1).

### 3. Hierarchy named 'AWS Observability' already exist
#### Error Message
`"errors":[{"code":"hierarchy:duplicate","message":"hierarchy named 'AWS Observability' already exist"}]`
#### Solution
Delete existing hierarchy and a create new one:<br/>

> 1. Get Hierarchy-id list of existing hierarchies and keep it noted.<br/>
>   `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies`
> 1. Delete the existing Hierarchy.<br/>
>   `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>`<br/>Learn [more](https://help.sumologic.com/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security) for apiendpoint.

Reviewer: ```suggestion
1. Get Hierarchy-id list of existing hierarchies and keep it noted.
   ```
   curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies
   ```
1. Delete the existing Hierarchy.
   ```
   curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>
   ```
   To find your `apiendpoint`, see [Sumo Logic endpoints](/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security).
```

> Delete existing hierarchy and a create new one:<br/>
> 1. Get Hierarchy-id list of existing hierarchies and keep it noted.<br/>
>   `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies`
> 1. Delete the existing Hierarchy.<br/>
>   `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>`<br/>Learn [more](https://help.sumologic.com/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security) for apiendpoint.

Reviewer: Format these snippets as [code blocks](https://help.sumologic.com/docs/contributing/style-guide/#code-blocks) (not code font) because it's snippets we want customers to copy/paste

```suggestion
Delete existing hierarchy and a create new one:
1. Get Hierarchy-id list of existing hierarchies and keep it noted.
  `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies`
1. Delete the existing Hierarchy.
  `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>`<br/>Learn [more](https://help.sumologic.com/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security) for apiendpoint.
```

Author: Comment incorporated

### 4. Cannot import name 'SumoLogic' from 'sumologic'
#### Error Message

> from sumologic import SumoLogic<br/>Import Error: cannot import name 'SumoLogic' from 'sumologic'<br/> `(/usr/local/lib/python3.10/site-packages/sumologic/__init__.py)`

Reviewer: All command snippets (for customers to copy/paste) must be formatted as [code blocks](https://help.sumologic.com/docs/contributing/style-guide/#code-blocks) (not code font).

#### Solution

> The package is [sumologic-sdk](https://pypi.org/project/sumologic-sdk/) and install it for AWS observability solution using the following command:<br/>`pip install sumologic-sdk`

Reviewer: Format this as [code block](https://help.sumologic.com/docs/contributing/style-guide/#code-blocks) (not code font) because it's a snippet we want customers to copy/paste

Author: Comment incorporated.

| `email_notifications`	Email Notifications to be sent by the alert. | `[ ]` |
| `connection_notifications` | Connection Notifications to be sent by the alert. | `[ ]` |
| `parent_folder_id` | The folder ID is automatically generated. Do not enter a value for this parameter. This is the folder ID to install the apps into. A folder using the provided name will be added in "apps_folder_name". If the folder ID is empty, apps will be installed in the Personal folder. | Ignore this parameter. |

## Troubleshooting

> This section provides information on how to troubleshoot failures while deploying AWS observability solution using Terraform.

Reviewer: ```suggestion
This section provides information on how to troubleshoot failures while deploying our AWS Observability solution using Terraform.
```

### Python command not found
#### Error Message

> `python source-module/attach_fields_to_source.py`<br/>Python: command not found

Reviewer: This is how the error message is displaying on localhost. Was this intentional?
<img width="300" alt="Screenshot 2023-05-17 at 2 06 50 PM" src="https://github.com/SumoLogic/sumologic-documentation/assets/56411016/92a553ec-af22-41d2-bd1c-d9349e81dad0">

Or should it be like this, all in one console message?
<img width="300" alt="Screenshot 2023-05-17 at 2 06 24 PM" src="https://github.com/SumoLogic/sumologic-documentation/assets/56411016/b163c156-3dbb-4a63-b1f1-c9b77a2a56c4">

Author: @kimsauce thanks for this. I feel its better we keep the error message in a console as you suggested.

Author: Fixed.

#### Solution
Identify and replace `python` with `python3` in [source-module/update_sources.tf](https://github.com/SumoLogic/sumologic-solution-templates/blob/AWSO_FY23Q4_Release/aws-observability-terraform/source-module/update_sources.tf#L12).

### Module not found
#### Error Message

> `Local-exec provisioner error`<br/>Module Not Found Error: No Module named ‘sumologic’

Reviewer: same question as https://github.com/SumoLogic/sumologic-documentation/pull/2322/files#r1197056237

Author: Fixed.

#### Solution
Verify you configured [Sumo Logic provider](https://github.com/SumoLogic/sumologic-solution-templates/blob/AWSO_FY23Q4_Release/aws-observability-terraform/providers.tf#L1).

### Hierarchy named 'AWS Observability' already exist
#### Error Message
`"errors":[{"code":"hierarchy:duplicate","message":"hierarchy named 'AWS Observability' already exist"}]`
#### Solution
Delete existing hierarchy and a create new one:<br/>
1. Get Hierarchy-id list of existing hierarchies and keep it noted.<br/>

>   ```sql
>     `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies`
>   ```
> 1. Delete the existing Hierarchy. Learn [more](https://help.sumologic.com/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security) for apiendpoint.<br/>
>   ```sql
>     `curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>`
>   ```

Reviewer: ```suggestion
  ```sql
    curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X GET https://<apiendpoint>/api/v1/entities/hierarchies
  ```
1. Delete the existing Hierarchy. Learn [more](https://help.sumologic.com/docs/api/getting-started/#sumo-logic-endpoints-by-deployment-and-firewall-security) for apiendpoint.<br/>
  ```sql
    curl -s -H 'Content-Type: application/json' --user <accessid>:<accesskey> -X DELETE https://<apiendpoint>/api/v1/entities/hierarchies/<hierarchyid>
  ```
```

### Cannot import name 'SumoLogic' from 'sumologic'
#### Error Message

> from sumologic import SumoLogic<br/>Import Error: cannot import name 'SumoLogic' from 'sumologic'<br/> `(/usr/local/lib/python3.10/site-packages/sumologic/__init__.py)`

Reviewer: same question as https://github.com/SumoLogic/sumologic-documentation/pull/2322/files#r1197056237

Author: Fixed.

#### Solution
The package is [sumologic-sdk](https://pypi.org/project/sumologic-sdk/) and install it for AWS observability solution using the following command:
  ```sql
  pip install sumologic-sdk
  ```

### Argument named *managed_apps* is not expected
#### Error Message

> An argument named **managed_apps** is not expected here.<br/>`Error: Unsupported argument`<br/>`on .terraform/modules/account.sumo_observability.app-modules/alb_app.tf line 13, in module "alb_module":`<br/>`managed_apps = {`

Reviewer: same question as https://github.com/SumoLogic/sumologic-documentation/pull/2322/files#r1197056237

Author: Fixed.

#### Solution

> Refer the module available [here](https://github.com/SumoLogic/terraform-sumologic-sumo-logic-integrations/tree/master/sumologic). 

Reviewer: ```suggestion
Refer to [this module in GitHub](https://github.com/SumoLogic/terraform-sumologic-sumo-logic-integrations/tree/master/sumologic). 
```

Author: incorporated.

### Argument named *admin_mode* is not expected
#### Error Message

> An argument named **admin_mode** is not expected here.<br/>`Error: Unsupported argument`<br/>`on .terraform/modules/account.sumo_observability/provider.tf line 5, in provider "sumologic":`<br/>`admin_mode = var.sumologic_folder_installation_location == "Personal folder" ? false:true`

Reviewer: same question as https://github.com/SumoLogic/sumologic-documentation/pull/2322/files#r1197056237

Author: Fixed.

#### Solution

> Sumologic provider [version 2.10.0](https://github.com/SumoLogic/terraform-provider-sumologic/blob/master/CHANGELOG.md#2100-september-22-2021) onwards supports `admin_mode` <br/>Refer the `admin_mode` module available [here](https://registry.terraform.io/providers/SumoLogic/sumologic/latest/docs#authentication).

Reviewer: ```suggestion
Sumologic provider [version 2.10.0](https://github.com/SumoLogic/terraform-provider-sumologic/blob/master/CHANGELOG.md#2100-september-22-2021) onwards supports `admin_mode` <br/>Refer to the [`admin_mode` module](https://registry.terraform.io/providers/SumoLogic/sumologic/latest/docs#authentication).
```

### Invalid function argument
#### Error Message

> Error: Invalid function argument<br/>`on.terraform/modules/sumo-module.overview_app.overview_module/sumologic/sumologic.tf line 67, in resource "sumologic_content" "SumoLogicApps":`<br/>`67: config = file(each.value.content_json)`

Reviewer: same question as https://github.com/SumoLogic/sumologic-documentation/pull/2322/files#r1197056237

Author: Fixed.

## Troubleshooting

> This section provides information on how to troubleshoot failures while deploying AWS observability soluton using Terraform.

Reviewer: ```suggestion
This section provides information on how to troubleshoot failures while deploying AWS observability solution using Terraform.
```
